python/paddle/optimizer/optimizer.py

```diff
@@ -148,7 +179,13 @@ def __init__(self,
         self._dtype = None
         # Infer the dtype form parameter
         if self._parameter_list:
-            self._dtype = self._parameter_list[0].dtype
+            if isinstance(self._parameter_list[0], dict):
```

Comment:
do we need check whether `self._parameter_list` only hold one dict?

Reply:
done, thx